build: add docker files for local standalone compilation system environment

peterwillcn opened the pull request:
1. Unified local standalone compilation system environment
2. Easy tracking of issues within a unified system
3. Dockerfile includes the construction of two target images, Alpine and Ubuntu

Build:
```
docker build --rm --target alpine -t buildbot:alpine .
docker build --rm --target ubuntu -t buildbot:ubuntu .
```
Usage:
```
docker run --interactive --rm --tty --ulimit 'nofile=1024:262144' \
   --volume "$(pwd):/workdir" --workdir '/workdir' buildbot:alpine /bin/bash
```

Ansuel commented:
Nono i know and I already know alpine needs some work. I have an old branch. It's mainly patching tools to support alpine.

peterwillcn commented:
> Nono i know and I already know alpine needs some work. I have an old branch. It's mainly patching tools to support alpine.

This branch https://github.com/Ansuel/openwrt/commits/alpine-retry or https://github.com/Ansuel/openwrt/commits/alpine-container ?

Ansuel commented:
@peterwillcn correct but as you can see it's old stuff.

peterwillcn commented:
ping @ynezz

peterwillcn committed:
build: add docker file for local standards compilation

Unified local standalone compilation system environment
 for easy tracking of issues within a unified system.

Build:
docker build --rm --target alpine -t buildbot:alpine .
docker build --rm --target ubuntu -t buildbot:ubuntu .

Usage:
docker run --interactive --rm --tty --ulimit 'nofile=1024:262144' \
   --volume "$(pwd):/workdir" --workdir '/workdir' buildbot:alpine /bin/bash

Signed-off-by: xiao bo <peterwillcn@gmail.com>
